lib/adiwg/mdtranslator/internal/module_utils.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module AdiwgUtils
+   def self.reconcile_hashes(hashA, hashB)
+      # merges hashes by value "truthyness"
+      # the "falsely" values we're looking to replace
+      # are nil, false, and empty {} & []
+      # there shouldn't be any situation with 2 truthy values
+      hashA.merge(hashB) do |_, oldval, newval|
+         (oldval.respond_to?(:empty?) && oldval.empty?) || !oldval ? newval : oldval
+      end
+   end
+end
```

Comment:
Still not totally clear on what situation this would be used in. So we're examining 2 objects, and removing anything that's "falsy", and merging into one?

Reply:
yeah so i'm not stoked on this and I want to avoid having to use it ( or completely remove it ). the reason why I created it is because `Individual` and `Organization` modules use a [contact](https://github.com/GSA/mdTranslator/blob/9b46e132ee5fa1095285a9c10a4e187c97c528c3/lib/adiwg/mdtranslator/internal/internal_metadata_obj.rb#L302-L319) object to store information inside of. Both of these modules/elements can contain a `CI_Contact` element relying on the same `contact` internal object. In the case of `eMailList` in the `contact` internal object that needs to bubble up from the `address` which can exist inside a contact element. I was probably bombarded with these considerations and made this as a result. I'm gonna revisit this.

Reply:
I think the logic makes sense and it has tests so it clearly works... I think I would like it to throw or log an error if the unexpected happens, that is the object has multiple truthy values. What should happen then?